Common/SecurityIdentifier.cs

```diff
@@ -350,6 +350,35 @@ public SecurityIdentifier(string symbol, ulong properties, SecurityIdentifier un
         /// <returns>A new <see cref="SecurityIdentifier"/> representing the specified symbol today</returns>
         public static SecurityIdentifier GenerateEquity(string symbol, string market, bool mapSymbol = true, IMapFileProvider mapFileProvider = null)
         {
+            return GenerateWithFirstDate(symbol, market, SecurityType.Equity, mapSymbol, mapFileProvider);
+        }
+
+        /// <summary>
+        /// This implementation uses the configured <see cref="IMapFileProvider"/> via the <see cref="Composer.Instance"/>
+        /// </summary>
+        /// <param name="symbol">The symbol as it is known today</param>
+        /// <param name="market">The market</param>
+        /// <param name="securityType">The security type of the symbol</param>
+        /// <param name="mapSymbol">Specifies if symbol should be mapped using map file provider</param>
+        /// <param name="mapFileProvider">Specifies the IMapFileProvider to use for resolving symbols, specify null to load from Composer</param>
+        /// <returns>A new <see cref="SecurityIdentifier"/> representing the specified symbol today</returns>
+        public static SecurityIdentifier GenerateWithFirstDate(string symbol, string market, SecurityType securityType, bool mapSymbol = true, IMapFileProvider mapFileProvider = null)
+        {
+            var firstDate = DefaultDate;
+            Func<DateTime, string, string, SecurityIdentifier> GenerateSid;
+
+            switch (securityType)
```

Comment:
This is cute, but seems unnecessary. This check could be moved to the end of the method and be replaced w/ the actual invocation of the method instead of having to store it in a `Func<DateTime, string, string SecurityIdentifier>`

In fact, this entire `switch` statement can be moved to the end of the method and each `case` would turn into `return GenerateXXX( params )`. I think this would be more explicit and remove this layer of indirection.